extensions/wikia/WallNotifications/WallNotificationEntity.class.php

```diff
@@ -78,174 +127,202 @@ public function getUniqueId() {
 		}
 	}
 
-	public function getId() {
-		return $this->id;
+	/**
+	 * This method calls out to the wiki given by $wikiId to get revision data, since
+	 * this data cannot be gathered locally if $wikiId != $wgCityId
+	 *
+	 * @param int $revId
+	 * @param int $wikiId
+	 *
+	 * @return bool
+	 */
+	public function loadDataFromRevIdOnWiki( $revId, $wikiId, $useMasterDB = false ) {
+		$dbName = WikiFactory::IDtoDB( $wikiId );
+		$params = [
+			'controller' => 'WallNotifications',
+			'method' => 'getEntityData',
+			'revId' => $revId,
+			'useMasterDB' => $useMasterDB,
+		];
+
+		$response = ApiService::foreignCall( $dbName, $params, ApiService::WIKIA );
+		if ( !empty( $response['status'] ) && $response['status'] == 'ok' ) {
+			$this->parentTitleDbKey = $response['parentTitleDbKey'];
+			$this->msgText = $response['msgText'];
+			$this->threadTitleFull = ['threadTitleFull'];
+			$this->data = ['data'];
+
+			return true;
+		}
+
+		return false;
 	}
 
-	public function loadDataFromRev( Revision $rev, $wikiId, $master = false ) {
-		global $wgSitename;
+	/**
+	 * @param int $revId
+	 * @param bool $userMasterDB
+	 *
+	 * @return bool
+	 */
+	public function loadDataFromRevId( $revId, $userMasterDB = false ) {
+		$rev = Revision::newFromId( $revId );
+		if ( empty( $rev ) ) {
+			return false;
+		}
 
-		$this->id = $rev->getId() . '_' .  $wikiId;
+		return $this->loadDataFromRev( $rev, $userMasterDB );
+	}
 
-		$wm = WallMessage::newFromTitle( $rev->getTitle() ); /* @var $wm WallMessage */
+	/**
+	 * @param Revision $rev
+	 * @param bool $useMasterDB
+	 *
+	 * @return bool
+	 */
```

Comment:
This method used to be really enormous.  I tried to organize it logically and move some things into smaller methods.  Its still a bit cumbersome however, due to the large number of data members stuffed into ```$data```